### PR TITLE
Add tests for tools & prototypes for wrappers tests

### LINKED ISSUE #4: Write prototype for wrapper tests
**Is your feature request related to a problem? Please describe.**
We need test for the Snakemake wrappers. At least, we would like to have smoke test that test whether the wrappers are work in principle.

**Describe the solution you'd like**
The current state of the art of running snakemake wrappers can be seen in the "official" wrappers repository here:

- https://github.com/snakemake/snakemake-wrappers/blob/master/test.py

Essentially they are just smoke tests with predefined `Snakefile`s, which is fine. We should have the same.

**Describe alternatives you've considered**
Using the official snakemake wrappers gives us an upstream dependency that we don't want. Further, we have some extended aims here that includes making copies of the wrapper files etc. and we don't want to discuss/convince the upstream snakemake wrappers community.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by holtgrewe**
This adds back the tests for the tools below `snappy_wrappers`.

It also creates tests for the wrappers, losely based on the tests for the "official" snakemake wrappers repository. The "containerized" running of wrappers is broken (probably also upstream in the official snakemake wrappers repository).

closes #4

### CODE REVIEW

**Comment by holtgrewe** [comment]
@eudesbarbosa this is ready for review now

**Comment by eudesbarbosa** [comment]
**Tests:** look good, but I cannot make depth comments as I'm a still getting used to the code.

**Imports:** There are some unused imports (see codacy results). Also, do we have a style defined? If not I would suggest going for something like:
- built-in modules
- third-party modules
- project modules

**Test structure:** Again, if no style yet:
- imports
- global vars
- fixtures
- tests

File: tests/snappy_wrappers/tools/test_bed_filter_jaccard.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def test_bed_filter_jaccard_subtract(capsys, input_txt_fake_fs, mocker):
+    mocker.patch("argparse.open", input_txt_fake_fs.open, create=True)
+    mocker.patch("argparse._os", input_txt_fake_fs.os)
+
+    bed_filter_jaccard.main(["--input-file", "/work/input.txt", "--operation", "subtract"])
+
+    out, err = capsys.readouterr()
+    assert (
```
Comment (eudesbarbosa) [inline tests/snappy_wrappers/tools/test_bed_filter_jaccard.py:33]: Cosmetics...
```
expected_out = textwrap.dedent(
            """
    2\t1\t200\t.\t-1\t-1\t0
    """
        ).lstrip()
    )
assert out == expected_out
```

File: tests/snappy_wrappers/tools/test_bed_filter_jaccard.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def test_bed_filter_jaccard_intersect(capsys, input_txt_fake_fs, mocker):
+    mocker.patch("argparse.open", input_txt_fake_fs.open, create=True)
+    mocker.patch("argparse._os", input_txt_fake_fs.os)
+
+    bed_filter_jaccard.main(["--input-file", "/work/input.txt", "--operation", "intersect"])
+
+    out, err = capsys.readouterr()
+    assert (
```
Comment (eudesbarbosa) [inline tests/snappy_wrappers/tools/test_bed_filter_jaccard.py:51]: Cosmetics...
```
expected_out = textwrap.dedent(
            """
   1\t1\t100\t1\t10\t110\t90
    """
        ).lstrip()
    )
assert out == expected_out
```

**Review by eudesbarbosa — COMMENTED** [review]
**Tests:** look good, but I cannot make depth comments as I'm a still getting used to the code.

**Imports:** There are some unused imports (see codacy results). Also, do we have a style defined? If not I would suggest going for something like:
- built-in modules
- third-party modules
- project modules

**Test structure:** Again, if no style yet:
- imports
- global vars
- fixtures
- tests

File: tests/snappy_wrappers/tools/test_bed_filter_jaccard.py (hunk 1 above)
Comment (holtgrewe) [reply tests/snappy_wrappers/tools/test_bed_filter_jaccard.py:33]: agreed, the post predates `black`...

File: tests/snappy_wrappers/tools/test_bed_filter_jaccard.py (hunk 2 above)
Comment (holtgrewe) [reply tests/snappy_wrappers/tools/test_bed_filter_jaccard.py:51]: agreed, the post predates `black`...

**Comment by holtgrewe** [comment]
Agreed to the proposed style. I think the review should be limited to the wrapper tests here as I just copied over the other tests. They will need some rework.

Regarding import style check, it would be nice to have this done automatically but I don't know which tools if any support this.

**Comment by holtgrewe** [comment]
Also, Github currently wants you to approve the change to let me commit it cleanly.

**Review by eudesbarbosa — APPROVED** [review]
LGTM!

**Pull request merged**